builders/reservationBuilder.go

```diff
@@ -0,0 +1,205 @@
+package builders
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/tfexplorer"
+	"github.com/threefoldtech/tfexplorer/client"
+	"github.com/threefoldtech/tfexplorer/models/generated/workloads"
+	wrklds "github.com/threefoldtech/tfexplorer/pkg/workloads"
+	"github.com/threefoldtech/tfexplorer/schema"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/crypto"
+)
+
+// ReservationBuilder is a struct that can build reservations
+type ReservationBuilder struct {
+	Reservation workloads.Reservation
+	Bcdb        *client.Client
+	Mainui      *tfexplorer.UserIdentity
+	assets      []string
+	seedPath    string
+	dryRun      bool
+}
+
+// NewReservationBuilder creates a new ReservationBuilder
+func NewReservationBuilder(bcdb *client.Client, mainui *tfexplorer.UserIdentity) *ReservationBuilder {
+	reservation := workloads.Reservation{}
+	reservation.Epoch = schema.Date{Time: time.Now()}
+	return &ReservationBuilder{
+		Reservation: reservation,
+		Bcdb:        bcdb,
+		Mainui:      mainui,
+	}
+}
+
+// LoadReservationBuilder loads a reservation builder based on a file path
+func LoadReservationBuilder(reader io.Reader) (*ReservationBuilder, error) {
+	reservation := workloads.Reservation{}
+	err := json.NewDecoder(reader).Decode(&reservation)
+	if err != nil {
+		return &ReservationBuilder{}, err
+	}
+
+	return &ReservationBuilder{Reservation: reservation}, nil
+}
+
+// Save saves the reservation builder to an IO.Writer
+func (r *ReservationBuilder) Save(writer io.Writer) error {
+	err := json.NewEncoder(writer).Encode(r.Reservation)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+// Build returns the reservation
+func (r *ReservationBuilder) Build() workloads.Reservation {
+	return r.Reservation
+}
+
+// Deploy deploys the reservation
+func (r *ReservationBuilder) Deploy() (wrklds.ReservationCreateResponse, error) {
```

Comment:
I'm not sure this belongs to the builder itself. This is more about what to do with the reservation then how to build it.

Reply:
Where do you suggest we move this logic? Back to the tfuser provision command?